core/src/main/java/org/apache/iceberg/SnapshotManager.java

```diff
@@ -50,25 +51,34 @@ protected ManageSnapshots self() {
 
   @Override
   protected String operation() {
-    // snapshotOperation is used by SnapshotProducer when building and writing a new snapshot for cherrypick
-    Preconditions.checkNotNull(snapshotOperation, "[BUG] Detected uninitialized operation");
+    //TODO: We are not setting this var during rollbacks. What should its value be during rollbacks?
```

Comment:
A rollback doesn't create a snapshot, it uses an existing one. In that case, `operation` is never called to create summary metadata. That's why there was the bug check here.

Reply:
So in that case, when we call `createSnapshotEvent` how should we handle the case when we want to call the `operation()` method.

Reply:
I think we should do what `snapshotId` does and report the operation of the picked snapshot. That should be safe, since the operation won't change. We'll have to remove this bug check and set the operation in all cases. So it would be basically what you have here, but with the TODO items finished.

Reply:
If we purposely leave it un-initialized, like in case of rollbacks. Is this then really a bug? Can we just return NULL value?

Reply:
I guess it depends on what you want the update event to contain. If that event reports the rollback, it could omit the operation or it could have its own logic for reporting the operation of the commit that was rolled back to.

Reply:
Sorry, to answer your question about "is this really a bug": it's a bug if we don't initialize it _and_ call to get the operation for some purpose. That's why it's a bug check.

Reply:
I think, In rollback's case, I can override the `updateEvent()` method  to not return a `CreateSnapshotEvent` object.

Reply:
That sounds good to me!